source/robot_model/src/Model.cpp

```diff
@@ -244,9 +244,85 @@ std::vector<state_representation::CartesianPose> Model::forward_geometry(
   return this->forward_geometry(joint_state, frame_ids);
 }
 
-state_representation::JointPositions Model::inverse_geometry(const state_representation::CartesianState&) const {
-  // TODO
-  return state_representation::JointPositions();
+StateRepresentation::CartesianPose Model::forward_geometry(const StateRepresentation::JointState& joint_state,
+                                                           std::string frame_name) {
+  if (frame_name.empty()) {
+    // get last frame if none specified
+    frame_name = this->robot_model_.frames.back().name;
+  }
+  return this->forward_geometry(joint_state, std::vector<std::string>{frame_name}).front();
+}
+
+StateRepresentation::JointPositions Model::inverse_geometry(const StateRepresentation::CartesianState& desired_cartesian_state,
+                                                            const StateRepresentation::JointState& current_joint_state,
+                                                            std::string frame_name, const double& tolerance, const int& max_number_of_iteration) {
+
+  if (frame_name.empty()) {
+    // get last frame if none specified
+    frame_name = this->robot_model_.frames.back().name;
+  }else {
+    // throw error if specified frame does not exist
+    if (!this->robot_model_.existFrame(frame_name)) { throw (Exceptions::FrameNotFoundException(frame_name)); }
+  }
```

Comment:
There are multiple formatting issues here (missing whitespace before `}else`, inline code block after an if statement instead of new line, extra space after `throw (`), in addition you should use `else if {` directly over `else { if {`.

See also the [style guide](https://wiki.epfl.ch/lasa/cpp-style-guide) and the new [contribution guide](https://github.com/epfl-lasa/control_libraries/blob/develop/CONTRIBUTING.md).